### ISSUE TITLE
Add a token status API which can be accessed without osio token

### ISSUE THREAD

**Issue by sbose78** [fabric8-services/fabric8-auth#345]
Proposal : 

- `GET /api/token/status/USERID?for=github` 
- `GET /api/user/USERID/token/status?for=github` 

Or add the info about OSO+ Github to `GET /api/user/USERID` ?

Or, we could protect it by a special service-account only for this purpose.

Note: none of these would return the actual token.

@alexeykazakov what do you think ?

**Comment by alexeykazakov** [comment]
Not sure we want to share this info with other users.. But maybe... Let's hold on until we have a usecase for such endpoint.

**Comment by sbose78** [comment]
>  we could protect it by a special service-account only for this purpose.
This would be a strong use-case for 'support engineers' in the near future.

**Comment by yzainee** [comment]
@sbose78 
we have a requirement where we have the git repo and we need the user's GH token to perform some operations on the repo. We would need some API where we provide the repo and service token and we get back the user's GH token

**Comment by sbose78** [comment]
@yzainee , Would you also provide the OSiO user ID whose corresponding Github token is needed?

**Comment by yzainee** [comment]
@sbose78 No, We wont have the OSIO user Id. @aslakknutsen Is there a way we can get the user id from the git repo (url) ?

**Comment by sbose78** [comment]
@yzainee Will all github URLs have a 1-o-1 mapping with osio users ?

**Comment by yzainee** [comment]
@sbose78 In case of multi collaborators this might not be true, but we still would need the GH token of any one of them to perform the operations on the repo

**Comment by yzainee** [comment]
Scenario:
We need to clone the repo, modify some files (pom.xml or req.txt etc) and then create a PR on the repo. This is the next step for the newer versions notification where, apart from sending the email, we also create a PR with the files changed with the newer versions

**Comment by miteshvp** [comment]
@sbose78 - In case of multiple collaborators, we can rely on admin of the workspace to create the PR. Of course, if Platform team can support this. Else, we can use any collaborator's Git token to open a PR. Shoubhik, do you see any concerns?

**Comment by sbose78** [comment]
Thanks @miteshvp @yzainee : 

I have a couple of concerns

1. If the codebase against which we want to open a PR is `fabric8-services/fabric8-notifications` , how would we identity whose Github token to use?

2. Assuming we do figure out whose openshiftio space that belongs to, I think this is what you are proposing: 
- Lookup the user's github token for a codebase associated with the user's space:
- Open a Pull request in the user's name.

I would feel very uncomfortable if we use the user's github token to do something on a user's public profile/repo when the user is not interacting with the system, and is away! And then open PRs in her name even though *she* did not open it :) 

Could we instead:
1. Send out an email notification with a 'Create PR' link, or show in the openshiftio UI.
2. When the user clicks on it, our service calls the Github Pull Requests API to create a PR on the user's behalf. Pretty much the same way as 'Create workitem'.

**Comment by miteshvp** [comment]
> I would feel very uncomfortable if we use the user's github token to do something on a user's public profile/repo when the user is not interacting with the system

@sbose78 - We will only be creating a PR if a user has permitted us on their behalf (based on some configs on profile page or workspaces page). If not, we will not do anything with the PR.

**Comment by yzainee** [comment]
@sbose78 The user would have enabled the notifications alerts on OSIO so it shouldnt be a problem if we are raising a PR as a part of the notification. Note here, that we are sending an email anyways, this is an enhancement on it. 
you can also see the dependabot features which does similar stuff. Check out the PRs it creates in this sample repo https://github.com/yzainee/hellovertx11

**Comment by yzainee** [comment]
@sbose78 https://github.com/openshiftio/openshift.io/issues/4284
There is a request for enhancements that we would need for the enable/disable of this feature.

**Comment by miteshvp** [comment]
@sbose78 - Concerns are valid. We were just giving it a thought on an approach. Currently, when launcher sets up a git repo, it uses git-id `openshift.io` like a bot. We could use similar mechanism where we get an approval from user from their codebases page to allow us to collaborate on the repo. If a user opts-in, we could use analytics bot to create the PR. That way, user as well as other collaborators would be aware of who created that PR.
